tests/test_butler.py

```diff
@@ -868,7 +871,9 @@ def testPutTemplates(self):
         butler.registry.registerDatasetType(DatasetType("metric2", dimensions, storageClass))
         butler.registry.registerDatasetType(DatasetType("metric3", dimensions, storageClass))
 
-        dataId1 = {"instrument": "DummyCamComp", "visit": np.int64(423)}
+        # Check that the dimension system can handle a numbers.Integral type
+        np_int = np.int64(423) if np else 423
+        dataId1 = {"instrument": "DummyCamComp", "visit": np_int}
```

Comment:
This is a bit different from the "skip if absent" strategy described in the commit message; in particular, somebody running this test without numpy will not realize there was a fallback. Why not use `@unittest.skipUnless` (with the added advantage that the current test will be factored into smaller pieces)?